lib/Lib.thy

```diff
@@ -2269,6 +2269,10 @@ lemma in_set_zip2:
   "(x, y) \<in> set (zip xs ys) \<Longrightarrow> y \<in> set ys"
   by (erule in_set_zipE)
 
+lemma set_zip_helper:
```

Comment:
This is a pretty generic name to go into lib; this looks something like a `in_set_zipD` based on the names of lemmas above

Reply:
+1 for `in_set_zipD`

Reply:
This is an old lemma that I moved to `Lib` because it appears in a few places and if I hadn't been moved to `Lib`, I would need to state it twice, due to the import structure not being linear. So I could rename it and then do one separate commit to use the new name in the proofs, and then fix up the name in the main commit

Reply:
And this lemma with the name `set_zip_helper` occurs on `master`, too. Considering that we want `Lib` to be as similar as possible between `master` and `rt`, I suppose I'll have to make a PR for `master` moving this lemma and renaming it...

Reply:
Good idea. It doesn't look big, so it'll go quickly.